test/e2e/e2e_suite_test.go

```diff
@@ -0,0 +1,38 @@
+package e2e
+
+import (
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1alpha1 "github.com/operator-framework/operator-controller/api/v1alpha1"
+)
+
+var (
+	cfg *rest.Config
+	c   client.Client
+)
+
+func TestE2E(t *testing.T) {
+	RegisterFailHandler(Fail)
+	SetDefaultEventuallyTimeout(1 * time.Minute)
+	SetDefaultEventuallyPollingInterval(1 * time.Second)
+	RunSpecs(t, "E2E Suite")
+}
+
+var _ = BeforeSuite(func() {
+	cfg = ctrl.GetConfigOrDie()
```

Comment:
Could these tests go under [suite_test.go](https://github.com/operator-framework/operator-controller/blob/main/controllers/suite_test.go)? Just because we have the test framework already scaffolded by the tool?

Reply:
Just saw Joe's comment https://github.com/operator-framework/operator-controller/pull/81#discussion_r1061098399, please feel free to ignore my review :)

Only one nit (we could do in a follow up), we could use envtest even for the unit (case 1) tests (though it is mainly intended for integration) if we foresee the interaction of multiple individual components, and would have to test "Reconcile" anyway. This in the case (1) where we need deppy entity source, and essentially mock the functioning of controller. This is just because previously we had faced version incompatibility issues by using locally available binaries and had a hard time figuring the cause.

Reply:
No worries, I have created an issue to follow up https://github.com/operator-framework/operator-controller/issues/86. We can start using envtest when we get to writing tests for reconciler depending on the complexity of those. Till then its good :)